pkcs11/src/new/mod.rs

```diff
@@ -83,20 +106,20 @@ mod tests {
         flags.set_rw_session(true).set_serial_session(true);
 
         // open a session
-        let session = pkcs11.open_session(&slot, flags).unwrap();
+        let session = pkcs11.open_session_no_callback(&slot, flags).unwrap();
 
         let pin = String::from("123456");
 
         // log in the session
-        pkcs11.login(&session, UserType::User, pin).unwrap();
+        pkcs11.login(&session, UserType::User, &pin).unwrap();
 
         // get mechanism
         let mechanism = Mechanism::RsaPkcsKeyPairGen;
 
-        let mut attr_true: u8 = 1;
-        let mut attr_true1: u8 = 1;
-        let mut attr_false: u8 = 0;
-        let mut public_exponent: Vec<u8> = vec![0x00, 0x01, 0x00];
+        let mut attr_true = pkcs11_sys::CK_TRUE;
+        let mut attr_true1 = pkcs11_sys::CK_TRUE;
+        let mut attr_false = pkcs11_sys::CK_FALSE;
```

Comment:
Didn't notice these before, but do you need different vars for these? I remember that the attributes that take in a "bool" (`CK_TRUE`, `CK_FALSE`) didn't have a problem with references/pointers. Doing just
```
CK_ATTRIBUTE::new(CKA_VERIFY).with_bool(&CK_TRUE)
```
was completely safe in Parsec. Seeing these here made me realise that `Attributes` isn't fully abstracted - for the ones that just need a boolean, we should make them take in a Rust `bool` and create the `CK_ATTRIBUTE` as above from them.

Reply:
Agreed! We need to complete the `Attribute` enum so that it only contains Rust native types.